docs/the-new-architecture/using-codegen.md

```diff
@@ -0,0 +1,199 @@
+
+<!-- TODO: add links -->
+
+## Configuring **Codegen**
+
+**Codegen** can be configured in your app by modifying the `package.json` file. **Codegen** is controlled by a custom field called `codegenConfig`.
+
+```json title="package.json"
+  "codegenConfig": {
+    "name": "<SpecName>",
+    "type": "<types>",
+    "jsSrcsDir": "<source_dir>",
+    "android": {
+      "javaPackageName": "<java.package.name>"
+    }
+  },
+```
+
+You can add this snippet to your app and customise the various fields:
+
+- `name:` This is the name that will be used to create files containig your specs. The convention is that it should have the suffix `Spec`, but this is not mandatory.
+- `type`: the type of code we need to generate. Allowed values are `modules`, `components`, `all`.
+  - `modules:` use this value if you only need to generate code for Turbo Native Modules.
+  - `components:` use this value if you only need to generate code for Native Fabric Components.
+  - `all`: use this value if you have a mixture of components and modules.
+- `jsSrcsDir`: this is the root folder where all your specs live.
+- `android.javaPackageName`: this is an android specific setting to let **Codegen** generate the files in a custom package.
+
+When **Codegen** runs, it searches among all the dependencies of the app, looking for JS files that respects some specific conventions, and it generates the required code:
+
+- Turbo Native Modules requires that the spec files are prefixed with `Native`. For example, `NativeLocalStorage.ts` is a valid name for a spec file.
+- Native Fabric Components requires that the spec files are suffixed with `NativeComponent`. For example, `WebViewNativeComponent.tx` is a valid name for a spec file.
+
+## Running **Codegen**
+
+The rest of this guide assumes that you have a Native Turbo Module and/or a Native Fabric Component already set up in your project. We also assume that you have valid specification files in the `jsSrcsDir` specified in the `package.json`.
+
+### Android
+
+**Codegen** for Android is integrated with the React Native Gradle Plugin (RNGP). The RNGP contains a task that can be invoked that reads the configurations defined in the `package.json` file and execute **Codegen**. To run the gradle task, first navigate inside the `android `folder of your project. Then run:
+
+```bash
+./gradlew generateCodegenArtifactsFromSchema
+```
+
+This task invokes the `generateCodegenArtifactsFromSchema` command on all the the imported projects of the app (the app and all the node modules which are linked to it). It generates the code in the corresponding `node_modules/<dependency>` folder. So, for example, if you have a Fabric Native Component whose node module is called `my-fabric-component`, the generated code is located in the `SampleApp/node_modules/my-fabric-component/android/build/generated/source/codegen` path. For the app, the code is generated in the `android/app/build/generated/source/codegen` folder.
```

Comment:
I do this a lot (I'm going to stop adding notes about it): 
- we should avoid starting a sentence with 'So'.  It doesn't add anything.
- be careful when using proper nouns: `gradle` vs "Gradle does X", `node` vs "whose Node module".
- avoid using "should", "would", "could" unless there is real doubt (stealing again from Shrunk & White).

<img src="https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExaXNvdHplenhxczVvcHc1bjl3OWF4eGlwNXY2M292MjdyMTk3NW1mMiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/XtdEdIgbZRXKMfsrEL/giphy.gif" />